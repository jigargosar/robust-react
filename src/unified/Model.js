// import 'dom-testing-library/extend-expect'
import {ListItemText, MenuItem, Paper} from 'material-ui'
// import {render} from 'react-testing-library'
// import {describe, expect, it, specs, storiesOf} from '../../../../facade'
import {storiesOf} from '../facade'
import {div, h} from '../hyper-script'

const ModelListItem = ({model}) => {
  return h(MenuItem, [h(ListItemText, model.name)])
}

// const createFakeModels = () => {
//   const chance = Chance(11)
//   return R.times(
//     (idx: number) => ({id: idx, name: chance.country({full: true})}),
//     3,
//   )
// }

storiesOf('Unified | Model', module)
  .addDecorator(story =>
    div(
      {
        style: {
          padding: '16px 0',
          backgroundColor: '#ddd',
          display: 'flex',
          justifyContent: 'center',
          alignItems: 'center',
        },
      },
      [div({style: {flex: 1, maxWidth: 300}}, [h(Paper, [story()])])],
    ),
  )
  .add('ListItem', () => {
    return h(ModelListItem, {model: {id: 0, name: 'Foo bar'}})
  })
