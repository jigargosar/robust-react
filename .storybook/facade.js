import {
  action as actionReal,
  linkTo as linkToReal,
  storiesOf as storiesOfReal,
} from '@storybook/react'

import x from 'expect'

import {
  after as afterReal,
  afterEach as afterEachReal,
  before as beforeReal,
  beforeEach as beforeEachReal,
  describe as describeReal,
  fit as fitReal,
  it as itReal,
  specs as specsReal,
  xdescribe as xdescribeReal,
  xit as xitReal,
} from 'storybook-addon-specifications'

export const storiesOf = storiesOfReal
export const action = actionReal
export const linkTo = linkToReal
export const specs = specsReal
export const describe = describeReal
export const it = itReal

export const beforeEach = beforeEachReal
export const afterEach = afterEachReal
export const before = beforeReal
export const after = afterReal

export const xit = xitReal
export const fit = fitReal
export const xdescribe = xdescribeReal

x.extend({
  toMatchSnapshot: () => ({message: 'Fake Snapshot For Storybook', pass: true}),
})

export const expect = x
