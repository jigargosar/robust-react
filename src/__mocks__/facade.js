// import {render} from 'react-testing-library'
import noCrashSnapshot from '../components/__tests__/helpers/noCrashSnapshot'

const currentEnv = jasmine.currentEnv_

export const describe = currentEnv.describe
export const it = currentEnv.it
export const beforeEach = currentEnv.beforeEach
export const afterEach = currentEnv.afterEach
export const xit = currentEnv.xit
export const xdescribe = currentEnv.xdescribe
export const fit = currentEnv.fit
export const after = () => {}
export const before = () => {}
export const expect = global.expect

export const action = () => {}
export const linkTo = () => {}
export const specs = spec => spec()

// export const snapshot = (name, story) => {
//   it(name, () => {
//     const {container} = render(story)
//     expect(container.firstChild).toMatchSnapshot()
//   })
// }

export const storiesOf = kind => ({
  add(name, func) {
    describe(kind, () => {
      describe(name, () => {
        it('should render without crashing', () => {
          func()
        })

        it('should match snapshot', () => {
          // const {container} = render(func())
          // expect(container.firstChild).toMatchSnapshot()
          noCrashSnapshot(func)
        })
      })
    })

    return this
  },
  addDecorator() {
    return this
  },
})
