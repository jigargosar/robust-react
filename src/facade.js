import expect_ from 'expect'

expect_.extend({
  toMatchSnapshot: () => ({message: 'Fake Snapshot For Storybook', pass: true}),
})

export {default as expect} from 'expect'

export {default as jest} from 'jest-mock'
export {
  after,
  afterEach,
  before,
  beforeEach,
  describe,
  fit,
  it,
  specs,
  xdescribe,
  xit,
} from 'storybook-addon-specifications'

export {storiesOf} from '@storybook/react'
export {action} from '@storybook/addon-actions'
export {linkTo} from '@storybook/addon-links'
