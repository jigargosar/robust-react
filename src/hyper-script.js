import h from 'react-hyperscript'
import hh from 'hyperscript-helpers'

export const {div} = hh(h)


export {h}
