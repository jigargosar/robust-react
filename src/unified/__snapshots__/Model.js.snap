// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unified | Model ListItem should match snapshot 1`] = `
<li
  class="MuiButtonBase-root MuiListItem-root MuiListItem-default MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
  role="menuitem"
  tabindex="-1"
>
  <div
    class="MuiListItemText-root"
  >
    <h3
      class="MuiTypography-root MuiTypography-subheading MuiListItemText-primary"
    >
      Foo bar
    </h3>
  </div>
  <span
    class="MuiTouchRipple-root"
  />
</li>
`;
